### ISSUE TITLE
Expanded surface format support: surface_rgb8unorm

### ISSUE THREAD

**Issue by Igameart** [YoYoGames/GameMaker-Feature-Requests#1204]
### Is your feature request related to a problem?

reading an rgb8 buffer into GM takes more processing time than necessary. 

### Describe the solution you'd like

I'd like to see support for surface_rgb8unorm so I don't have to read every byte one after the other to convert an rgb8 buffer into a usable surface.

### Describe alternatives you've considered

Currently I am reading s16, followed by s8 for every pixel of data in my buffer. This is incredibly inefficient when dealing with upwards of 200,000 individual buffers that need loading.

**Comment by AtlaStar** [comment]
according to the docs, the default for a surface is the surface_rgb8unorm type....not sure what it is that you are actually wanting based on the request. I think clarification is needed to understand exactly what you mean.

Are you saying that using a buffer to set a surface _doesn't_ properly work with surface_rgb8unorm types without some extra work, or something else?

**Comment by Igameart** [comment]
This is incorrect. There's an "a" in there. There is currently no surface_rgb8unorm format.
![image](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/5240715/dc2612ea-0ac3-4fc4-9fc7-12f3480a0eb2)

**Comment by AtlaStar** [comment]
Ahhh, ok my brain literally read what you wrote as an rgba8unorm and not just a rgb8unorm.

I think the reason this isn't supported is because openGL _doesn't_ support that type. It has r8, rg8, rgba8, and even signed norm variants....but not a rgb8unorm format qualifier.

This is almost certainly has to do with the way the GPU registers optimally work, as every format that is supported is either a byte, two bytes, four bytes, etc, so that the number of bytes is some power of 2 so that data loads align to word sizes, or byte size as most processors can handle loading individual bytes.

So basically; you are gonna have to add an alpha channel to your buffer if you want more performant code, as I don't think gamemaker is keen to support exotic image formats and write abstractions that let you do those types of things when you really should just be including an alpha channel in your buffers to begin with...and if this is a matter of reading a stream of data into a buffer, then you either are stuck overwriting the data to be compliant, or using hacks like you are now.

**Comment by Igameart** [comment]
I have no way of injecting an alpha channel into the buffers as these are buffers compiled by another tool I'm deciphering unless I can write a custom version of that tool to include a blank alpha channel. Which may be possible? I'll find out I suppose.

**Comment by AtlaStar** [comment]
If the buffer doesn't contain any positioning data, you may be able to use a hack by loading the buffer using `vertex_create_buffer_from_buffer` and a custom vertex format that hijacks the 3d positioning type. Then instead of setting the buffer as a surface, use the position data as color data in a shader.

Would be a janky hack that only works in specific circumstances, but it may be a more performant choice than what you are dealing with now.

**Comment by Igameart** [comment]
The buffer is only a list of RGB values in sequential order with a predetermined width and height.

**Comment by AtlaStar** [comment]
Then the hack may just work for ya

**Comment by Igameart** [comment]
Feel free to ping me on discord. I'm the guy making the Quake thing ;P Lol

**Comment by pkraif** [comment]
Since 24bpp texture formats are not widely supported, we will be implementing a new function instead, using which you will be able to easily solve this issue. See https://github.com/YoYoGames/GameMaker-Feature-Requests/issues/1298.
